ValkeyProxy.md

```diff
@@ -0,0 +1,112 @@
+
+c. When the primary node processes a large key, it may fail to respond to other nodes for a long time, resulting in disconnection of the client
+
+d. When the number of connections to a node exceeds the maximum value, the client may not be able to connect to the primary node
+
+Therefore, we need a proxy to make the customer unaware of the above problems, and also to provide some additional Valkey services
+
+
+## Design Considerations  (Required)
+
+1.	Support all existing Valkey commands 
+
+2.	Shield the functional/API differences between cluster mode and primary-replica mode, so that clients do not need to care about the underlying operating mode when using it.
+
+        Now, If the client is directly connected to the server, then in primary-replica mode and cluster mode, the client may need to execute different code logic for the two modes in the following scenarios:
+        a. The client needs to save sharding metadata information in cluster mode.
+        b. When failover happens in these 2 different modes, the client needs to do different processing.
+        c. In cluster mode, if data migration occurs, the client must handle it.
+
+        So if the client is only connected to the Proxy instead of server node, then as long as the client is not connected inSentinel mode, then when the client sends a command, it does not need to pay too much attention to the underlying connection mode. The Proxy will handle the execution of different modes for the client.
+
+3.	Support multi-database mode even server runs as cluster mode
+
+4.	By using Proxy, clients have no perception for server failover, slot migration, adding/removing server nodes, proxy nodes and other exceptions.
+
+        Through Proxy, we can avoid the client being directly connected to the underlying server stage. All client connections, reading and writing request will only be aware of the Proxy, so the client does not need the underlying Valkey server or 
+how data is storaged. 
+
+        Therefore, no matter whether failover happens, re-electing the primary node, resharding after adding a server node, or even adding a Proxy node, clients don't need to worry about it. 
+
+        It greatly simplifies the programming complexity of the client, allowing the client to focus more on business logic without caring too much about changes in the underlying architecture.
+
+5.      Performance optimization
+
+        We can use Proxy to optimize in different ways for different clients. For example, we can support Pipeline technology on the Proxy side to improve the overall system response efficiency.
+
+        For some time-consuming or high CPU cost commands, we can make statistics in Proxy side, and for some results, such as min/max/avg/p95/p99 from memtier_benchmark or valkey_benchmark, clients can get timely statistical information.
+
+
+6.	Support audit log 
```

Comment:
Cool, this is a hook in my opinion

Reply:
On benefit implemented in proxy side is to retrieve this information from one point, user do not need to fetch the log in different nodes in the cluster mode.